test/jdk/java/awt/image/multiresolution/MultiDisplayTest.java

```diff
@@ -21,63 +21,106 @@
  * questions.
  */
 
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.Font;
+import java.awt.Frame;
+import java.awt.Graphics;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.image.BaseMultiResolutionImage;
+import java.awt.image.BufferedImage;
+
+import javax.swing.JFrame;
+
+import jdk.test.lib.Platform;
+import jtreg.SkippedException;
 
 /*
-  @test
-  @bug 8142861 8143062 8147016
-  @summary Check if multiresolution image behaves properly
-           on HiDPI + non-HiDPI display pair.
-  @author a.stepanov
-  @library /test/lib
-  @build jdk.test.lib.Platform
-  @run applet/manual=yesno MultiDisplayTest.html
-*/
-
-
-import java.applet.Applet;
-import java.awt.*;
-import java.awt.event.*;
-import java.awt.image.*;
+ * @test
+ * @bug 8142861 8143062 8147016
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame jdk.test.lib.Platform
+ * @requires (os.family == "windows" | os.family == "mac")
+ * @summary Check if multiresolution image behaves properly
+ *         on HiDPI + non-HiDPI display pair.
+ * @run main/manual MultiDisplayTest
+ */
 
-import jdk.test.lib.Platform;
+public class MultiDisplayTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             The test requires two-display configuration, where
+
+             - 1st display is operating in HiDPI mode;
+             - 2nd display is non-HiDPI.
+
+             In other cases please simply push "Pass".
+
+             To run test please push "Start".
+
+             Then drag parent / child to different displays and check
+             that the proper image is shown for every window
+             (must be "black 1x" for non-HiDPI and "blue 2x" for HiDPI).
 
-public class MultiDisplayTest extends Applet {
+             Please try to drag both parent and child,
+             do it fast several times and check if no artefacts occur.
+
+             Try to switch display resolution (high to low and back).
+
+             For Mac OS X please check also the behavior for
+             translucent windows appearing on the 2nd (non-active) display
+             and Mission Control behavior.
+
+             Close the windows.
+
+             In case if no issues occur please push "Pass", otherwise "Fail"
+            """;
 
     private static final int W = 200, H = 200;
```

Comment:
```suggestion
    private static final int W = 200;
    private static final int H = 200;
```
Java Coding Guidelines do not recommend having several declarations on the same line.

Reply:
Fixed. Thanks